tree/ntuple/v7/src/RNTupleMerger.cxx

```diff
@@ -197,7 +197,7 @@ struct RChangeCompressionFunc {
       sealConf.fElement = &fDstColElement;
       sealConf.fPage = &page;
       sealConf.fBuffer = fBuffer;
-      sealConf.fCompressionSetting = fMergeOptions.fCompressionSettings;
+      sealConf.fCompressionSetting = fMergeOptions.fCompressionSettings.value();
```

Comment:
After the initial check at lines 889-896 the compression settings must be set to a valid value, so we can access it with a simple `*` during the whole `Merge` function